core/domain/suggestion_registry.py

```diff
@@ -319,30 +318,124 @@ def accept(self, commit_message):
 
-        Args:
-            suggestion_dict: dict. The dict representation of the suggestion.
+class SuggestionAddQuestion(BaseSuggestion):
+    """Domain object for a suggestion of type SUGGESTION_TYPE_ADD_QUESTION."""
 
-        Returns:
-            SuggestionEditStateContent. The corresponding Suggestion domain
-                object.
+    def __init__( # pylint: disable=super-init-not-called
+            self, suggestion_id, target_id, target_version_at_submission,
+            status, author_id, final_reviewer_id,
+            change_cmd, score_category, last_updated):
+        """Initializes an object of type SuggestionAddQuestion
+        corresponding to the SUGGESTION_TYPE_ADD_QUESTION choice.
+        """
+        self.suggestion_id = suggestion_id
+        self.suggestion_type = suggestion_models.SUGGESTION_TYPE_ADD_QUESTION
+        self.target_type = suggestion_models.TARGET_TYPE_TOPIC
+        self.target_id = target_id
+        self.target_version_at_submission = target_version_at_submission
+        self.status = status
+        self.author_id = author_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = question_domain.QuestionChange(change_cmd)
+        self.score_category = score_category
+        self.last_updated = last_updated
+
+    def validate(self):
+        """Validates a suggestion object of type SuggestionAddQuestion.
+
+        Raises:
+            ValidationError: One or more attributes of the SuggestionAddQuestion
+                object are invalid.
+        """
+        super(SuggestionAddQuestion, self).validate()
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_QUESTION:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_QUESTION,
+                    self.get_score_type()))
+        if not isinstance(self.change_cmd, question_domain.QuestionChange):
+            raise utils.ValidationError(
+                'Expected change_cmd to be an instance of QuestionChange')
+
+        if not self.change_cmd.cmd:
+            raise utils.ValidationError('Expected change_cmd to contain cmd')
+
+        if (
+                self.change_cmd.cmd !=
+                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION):
+            raise utils.ValidationError('Expected cmd to be %s, obtained %s' % (
+                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION,
+                self.change_cmd.cmd))
+
+        if not self.change_cmd.question_dict:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain question_dict')
+
+        if 'question_state_data' not in self.change_cmd.question_dict:
+            raise utils.ValidationError(
+                'Expected question dict to contain question_state_data')
+
+        if 'language_code' not in self.change_cmd.question_dict:
```

Comment:
It's also got to be a valid language code, right?

Reply:
Done